test/shopify_app/login_protection_test.rb

```diff
@@ -7,6 +7,7 @@ class LoginProtectionController < ActionController::Base
   helper_method :shop_session
 
   before_action :login_again_if_different_shop, only: [:second_login]
+  around_action :shopify_session, only: [:with_a_shopify_session]
```

Comment:
interesting. Can we re-use index or will that break tests?. Can we rename the route `internal` or something to indicate that it is behind auth and not related to it